src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt

```diff
@@ -134,6 +134,13 @@ class Microservice(
                 executor
             ) { batch ->
                 transportMessageRouter.send(batch)
+
+                for (group in batch.groups) {
+                    for (message in group.messages)
+                        if (message is RawMessage)
+                            message.body.release()
+                }
+
```

Comment:
Maybe set a copy of buffer into body instead of release buffer in several places